apputils/containerized.py

```diff
@@ -18,107 +20,117 @@ def __init__(self):
-        self.client.containers.get(container_name).remove()
+                env_vars[field] = os.getenv(field)
+        result = self.generic_container_launch_attached(image, env_vars,
+                                                        env_expand)
         return result.replace('\n', '')
 
     def generate_firewall_graph(self, target):
-        image = ("docker.io/halotools/firewall-graph:%s"
-                 % self.fw_graph_ver)
-        container_name = "halo_firewall_graph"
-        environment = {
-            "HALO_API_KEY": self.halo_key,
-            "HALO_API_SECRET_KEY": self.halo_secret,
-            "HTTPS_PROXY": self.https_proxy,
-            "TARGET": target
-        }
-
-        # Remove the container by name if it still exists from a prior run.
-        try:
-            self.client.containers.get(container_name).remove()
-        except docker.errors.APIError:
-            pass
-        result = self.client.containers.run(image, name=container_name,
-                                            detach=False,
-                                            mem_limit=self.mem_limit,
-                                            environment=environment)
-        self.client.containers.get(container_name).remove()
+        image = ("docker.io/halotools/firewall-graph:%s" % self.fw_graph_ver)
+        env_vars = {"TARGET": target}
+        env_expand = {"HALO_API_KEY": "HALO_API_KEY",
+                      "HALO_API_SECRET_KEY": "HALO_API_SECRET_KEY",
+                      "HTTPS_PROXY": "HTTPS_PROXY"}
+        result = self.generic_container_launch_attached(image, env_vars,
+                                                        env_expand, False)
         return result.replace('\n', '')
 
     def scans_to_s3(self, target_date, s3_bucket_name):
         image = ("docker.io/halotools/halo-scans-archiver:%s"
                  % self.scans_to_s3_ver)
-        container_name = "halo_scans_to_s3"
-        environment = {
-            "HALO_API_KEY": self.halo_key,
-            "HALO_API_SECRET_KEY": self.halo_secret,
-            "TARGET_DATE": target_date,
-            "AWS_S3_BUCKET": s3_bucket_name,
-            "AWS_ACCESS_KEY_ID": self.aws_key,
-            "AWS_SECRET_ACCESS_KEY": self.aws_secret,
-            "HTTPS_PROXY": self.https_proxy
-        }
-
-        # Remove the container by name if it still exists from a prior run.
-        try:
-            self.client.containers.get(container_name).remove()
-        except docker.errors.APIError:
-            pass
-        result = self.client.containers.run(image, name=container_name,
-                                            detach=False,
-                                            mem_limit=self.mem_limit,
-                                            environment=environment)
-        self.client.containers.get(container_name).remove()
+        env_vars = {"TARGET_DATE": target_date,
+                    "AWS_S3AWS_S3_BUCKET": s3_bucket_name}
```

Comment:
Typo: "AWS_S3AWS_S3_BUCKET" should be "AWS_S3_BUCKET"

Reply:
JyeCorrect > AutoCorrect  Fix coming right up.